apps/prairielearn/python/zygote.py

```diff
@@ -171,12 +171,21 @@ def worker_loop() -> None:
         # return the results. The caller should terminate us with a
         # SIGTERM.
         while True:
-            # wait for a single line of input
+            # Wait for a single line of input
             json_inp = sys.stdin.readline()
-            # unpack the input line as JSON
-            inp = json.loads(json_inp, parse_int=zu.safe_parse_int)
 
-            # get the contents of the JSON input
+            # Sometimes we seem to get an empty line, so we'll just ignore it.
+            if not json_inp.strip():
+                continue
```

Comment:
@mwest1066 based on the extra logging I added below, we were indeed randomly receiving newlines. I can't tell why that would be since we always write non-empty data immediately followed by a newline:

https://github.com/PrairieLearn/PrairieLearn/blob/ed34f9df2bb10f62c769ce16a50873468a8350be/apps/prairielearn/src/lib/code-caller/code-caller-native.ts#L262-L263

This change here resolved the errors. Are you good with this as a "hacky" fix to a problem I don't fully understand?

Reply:
I can't reproduce locally. This _could_ be causing problems in production; this would manifest as us losing workers and needing to restart them unnecessarily, which would increase latency. I guess we could check the logs to see if this is happening? I'm sort of inclined to add this to be extra safe?

Reply:
I also couldn't reproduce locally. I'm happy to add this to make sure prod is ok, but I just hate so much not knowing what the underlying cause is (and I know you do too).

Reply:
As do I! Any thoughts about how we could better investigate this?

Reply:
No good ideas at all 😭

I'm happy to merge this as some extra safety.